OMCompiler/Compiler/BackEnd/IndexReduction.mo

```diff
@@ -1387,6 +1389,44 @@ algorithm
   end match;
 end countStateCandidates;
 
+protected function countStateCandidatesWithNever
+"author: kabdelhak FHB 2019-07
+  count the number of states with stateSelect.never"
+  input BackendDAE.Var inVar;
+  input Integer inCount;
+  output BackendDAE.Var outVar;
+  output Integer outCount;
+algorithm
+  (outVar,outCount) := match inVar
```

Comment:
Why is `outVar` part of the match statement? It seems always be assigned to `inVar`, which isn't changed in the mach statement...

Reply:
> Why is `outVar` part of the match statement? It seems always be assigned to `inVar`, which isn't changed in the mach statement...

Like in the other case, i just copied the function `countStateCandidates` and adapted it a little. Does it have an impact (right now this function is not used but will be with one of my future commits so i could fix it then)?

Reply:
In general, it is bad code and makes it difficult to understand. I know there are a lot of places using this pattern which makes it unpleasant to read.